### PR TITLE
Add hibernate validator

### PR DESCRIPTION
**Pull request by sandrobonazzola**
Needed as build dependency by ovirt-engine

Signed-off-by: Sandro Bonazzola <sbonazzo@redhat.com>
Related-To: https://bugzilla.redhat.com/show_bug.cgi?id=2005978

### CODE REVIEW

File: ovirt-dependencies.spec.in [hunk 1]
```diff
@@ -19,6 +19,8 @@ BuildRequires:  maven >= 3.5.0
 BuildRequires:  maven-local
 BuildRequires:  maven-dependency-plugin
 BuildRequires:  tar
+BuildRequires:  bean-validation-api
+BuildRequires:  classmate
```
Comment (arso) [inline ovirt-dependencies.spec.in:23]: Where is this 'classmate' package available from ?  I was unable to find it.

**Review by arso — COMMENTED** [review]
Other than my 'classmate' related concern it looks  good. Before merging I would suggest to verify it by building ovirt-engine to make sure there are no missing dependencies with 'provided' scope (in such case the build will fail with some ClassDefNotFound exceptions)

Edit: I have just noticed the check failing because of the missing 'classmate' package.  It is available for centos7 (epel) so I guess it can be rebuilt, am I correct?

File: ovirt-dependencies.spec.in (hunk 1 above)
Comment (sandrobonazzola) [reply ovirt-dependencies.spec.in:23]: I built it: https://cbs.centos.org/koji/packageinfo?packageID=9000

**Comment by sandrobonazzola** [comment]
> Other than my 'classmate' related concern it looks good. Before merging I would suggest to verify it by building ovirt-engine to make sure there are no missing dependencies with 'provided' scope (in such case the build will fail with some ClassDefNotFound exceptions)

I need some help with the copyright notice and the verification of the package. I can handle the test with ovirt-engine build.

> 
> Edit: I have just noticed the check failing because of the missing 'classmate' package. It is available for centos7 (epel) so I guess it can be rebuilt, am I correct?

I took last Fedora 31 build and bumped to 1.3.4 as it was the version required by hibernate. Pushed to CBS.

File: conf.d/hibernate.conf [hunk 2]
```diff
@@ -0,0 +1,13 @@
+NAME="Hibernate Validator Aggregator"
+DESCRIPTION="Aggregator of the Hibernate Validator modules."
+HOME="http://hibernate.org/validator"
+LICENSE="ASL 2.0"
+COPYRIGHT=""
```
Comment (sandrobonazzola) [inline conf.d/hibernate.conf]: @arso @mwperina can you help here on the copyright part?

File: conf.d/hibernate.conf [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+COPYRIGHT_NOTE=''
+VERSION="6.0.18"
+ARTIFACT0="hibernate-validator.jar"
+ARTIFACT1="hibernate-validator.pom"
+ARTIFACT2="hibernate-validator-parent.pom"
+VERIFIER=""
+VERIFY_DATE=""
+OVIRT_VERSIONS="4.5"
```
Comment (sandrobonazzola) [inline conf.d/hibernate.conf:13]: also here about the verification part, anything else needed other than checking engine builds with this?

Comment (arso) [reply conf.d/hibernate.conf:13]: I think if the build passes we are good... but to be sure OST basic suite run would verified it in 100% (or  I can build rpm and run engine on VM myself)

File: conf.d/hibernate.conf (hunk 2 above)
Comment (arso) [reply conf.d/hibernate.conf]: I am wondering whether we could link to hibernate-validator GH: https://github.com/hibernate/hibernate-validator/blob/main/copyright.txt ?

File: conf.d/hibernate.conf (hunk 3 above)
Comment (sandrobonazzola) [reply conf.d/hibernate.conf:13]: If you have already everything set for the test please go ahead :-) I'm building a missing dep on el9.

Comment (sandrobonazzola) [reply conf.d/hibernate.conf:13]: Build passes. RPM build will fail later on at
```
Exception: Following dependencies were not resolved and requires cannot be generated. Either remove the dependencies from pom.xml or add proper packages to BuildRequires:
org.infinispan:infinispan-core:9.4.18.Final required by org.ovirt.engine.core:utils
org.jboss.modules:jboss-modules:1.1.1.GA required by org.ovirt.engine.core:common
```
going to add them in next PRs. If the PR looks good to you let's merge and move to next library.

File: conf.d/hibernate.conf (hunk 2 above)
Comment (sandrobonazzola) [reply conf.d/hibernate.conf]: done

File: conf.d/hibernate.conf (hunk 3 above)
Comment (sandrobonazzola) [reply conf.d/hibernate.conf:13]: seems to work

**Comment by sandrobonazzola** [comment]
Not needed anymore

**Pull request closed without merging**